resources/lib/player_listener.py

```diff
@@ -31,6 +31,26 @@ def _sanity_check_segments(segments):  # type: (Iterable[SponsorSegment]) -> boo
     return True
 
 
+def merge_overlapping_segments(segments): # type: (Iterable[SponsorSegment]) -> Iterable[SponsorSegment]
+    mergedsegments = []
+    itersegments = iter(segments)
+    curseg = next(itersegments)
+
+    for seg in itersegments: # type: SponsorSegment
+        if seg.start - curseg.end <= 0.5:
+            logger.debug("%s: merging into previous segment", seg)
+            curseg = curseg._replace(end = seg.end)
```

Comment:
this just overwrites the old end time, should we also modify the other values, for example the category?
As we merge different types into each other there is no single category which works here

Reply:
That's exactly the reason I didn't go for this approach. I want to preserve the metadata so we can report the segments (requires the uuid) and ideally show the user which categories were skipped (although I never got to this).
I considered adding a `MergedSponsorSegment` class that contains a list of segments and exposes the total start / end time, but the approach I ended up taking took less time to implement :P